core/domain/exp_services.py

```diff
@@ -1710,6 +1710,502 @@ def is_voiceover_change_list(change_list):
+    if backend_version_exploration.version == frontend_version:
+        return True
+    else:
+        # A complete list of changes from one version to another
+        # is composite_change_list.
+        composite_change_list = get_composite_change_list(
+            exp_id, frontend_version,
+            backend_version_exploration.version)
+
+        # Added_state_names: list(str). Names of the states added to the
+        # exploration from prev_exp_version to current_exp_version. It
+        # stores the latest name of the added state.
+        added_state_names = []
+
+        # Deleted_state_names: list(str). Names of the states deleted from
+        # the exploration from prev_exp_version to current_exp_version.
+        # It stores the initial name of the deleted state from
+        # pre_exp_version.
+        deleted_state_names = []
+
+        # New_to_old_state_names: dict. Dictionary mapping state names of
+        # current_exp_version to the state names of prev_exp_version.
+        # It doesn't include the name changes of added/deleted states.
+        new_to_old_state_names = {}
+
+        # Changed_properties: dict. List of all the properties changed
+        # according to the state and property name.
+        changed_properties = {}
+
+        # Changed_translations: dict. List of all the translations changed
+        # according to the state and content_id name.
+        changed_translations = {}
+
+        # properties_related_to_cust_args: List of the properties
+        # related (affected by or affecting) customization args. This list
+        # can be changed when any new property is added or deleted which
+        # affects or is affected by customization args.
+        properties_related_to_cust_args = ['solution', 'recorded_voiceovers',
+                                           'answer_groups',
+                                           'widget_customization_args']
+
+        # properties_related_to_answer_groups: List of the properties
+        # related (affected by or affecting) answer groups. This list
+        # can be changed when any new property is added or deleted which
+        # affects or is affected by answer groups.
+        properties_related_to_answer_groups = ['solution',
+                                               'recorded_voiceovers',
+                                               'answer_groups',
+                                               'widget_customization_args']
+
+        # properties_related_to_solution: List of the properties
+        # related (affected by or affecting) solution. This list
+        # can be changed when any new property is added or deleted which
+        # affects or is affected by solution.
+        properties_related_to_solution = ['solution', 'answer_groups',
+                                          'recorded_voiceovers',
+                                          'widget_customization_args']
```

Comment:
Aren't these the same list? If so, can we use a single list with an appropriate name.

Reply:
These lists are the same but they are for different properties so if in the future any property is added which is affected by solution only then in that case we'll add that property name in properties_related_to_solution only. So I don't think so we should keep the common list.

Reply:
Sounds good, can you please explicitly mention this in the comments so future devs don't combine the lists accidentally? Thanks!

Reply:
Explained it.